### PR TITLE
[docs] #174: Update documentation after RC-9 (LTS)

### LINKED ISSUE #140: Add a section on WASM validators
Add info about the chain of runtime validators.
Based on https://github.com/hyperledger/iroha/pull/2641.

(stable version)

### LINKED ISSUE #156: Update `FindTransactionsByAccountId`
Update [`FindTransactionsByAccountId`](https://hyperledger.github.io/iroha-2-docs/guide/advanced/queries.html#findtransactionsbyaccountid) after https://github.com/hyperledger/iroha/pull/2704 is merged

```
pub enum TransactionValue {
    /// Committed transaction
    Transaction(Box<VersionedSignedTransaction>),
    /// Rejected transaction with reason of rejection
    RejectedTransaction(Box<VersionedRejectedTransaction>),
}
```

### LINKED ISSUE #174: Update docs based on RC-9 code updates

### LINKED ISSUE #183: Add IROHA_GENESIS_ACCOUNT_PUBLIC_KEY and IROHA_GENESIS_ACCOUNT_PRIVATE_KEY
After latest changes in configuration it’s required to provide genesis account private and public keys through configuration or env variables:
```
IROHA_GENESIS_ACCOUNT_PUBLIC_KEY
IROHA_GENESIS_ACCOUNT_PRIVATE_KEY
```
See https://github.com/hyperledger/iroha/pull/2805/files#r986523562

This affects at least this page: https://hyperledger.github.io/iroha-2-docs/guide/advanced/running-iroha-on-bare-metal.html

### LINKED ISSUE #184: Update object structures
With less verbose json representation introduced in https://github.com/hyperledger/iroha/pull/2772, code examples need to be updated . Object hierarchy chapter as well.

For example, the change in domains from:
```
    "id": {
        "name": "wonderland"
    },
```
to
```
"id": "wonderland",
```

### LINKED ISSUE #186: Update docs on permission token definition (parameters)
Doc updates based on https://github.com/hyperledger/iroha/pull/2694

### LINKED ISSUE #187: Update client config example
Based on https://github.com/hyperledger/iroha/pull/2782 (see https://github.com/hyperledger/iroha/pull/2782/files#r986666649)

### LINKED ISSUE #188: Document listeners behaviour
Based on https://github.com/hyperledger/iroha/pull/2788: every listener is notified to shutdown on panic

### PR DESCRIPTION
**Pull request by outoftardis**
Signed-off-by: Ekaterina Mekhnetsova <mekkatya@gmail.com>

- Closes #156
- Closes #183
- Closes #186
- Closes #187
- Closes #140 
- Closes #188 
- Closes #184
- added info about configuration types (from condig.md in the main repo)
- updated references about different versions
- updated docker-compose file
- updated peer config page
- added a page with all config examples present (to avoid copy-pasting, configs are stored in json files and included in other md)

Closes #174

### CODE REVIEW

File: src/guide/configure/peer-configuration.md [hunk 1]
```diff
@@ -117,3 +138,13 @@ You can also set `LOG_FILE_PATH` to `/dev/stdout` if you want to use
 bunyan's logging facilities directly without saving the output.
 
 :::
+
+## Kura
+
+KURA is the "warehouse" engine of Iroha. The `BLOCK_STORE_PATH` specifies
```
Comment (appetrosyan) [inline src/guide/configure/peer-configuration.md:144]:
```suggestion
_Kura_ is the persistent storage engine of Iroha (Japanese for _warehouse_). The `BLOCK_STORE_PATH` specifies
```

File: src/guide/configure/intro.md [hunk 2]
```diff
@@ -6,6 +6,9 @@ the [peer configuration](./peer-configuration.md) and
 [client configuration](./client-configuration.md) file needed to interact
 with the blockchain.
 
+Read about the difference in underlying [types for configuration options](./configuration-types.md).
```
Comment (WRRicht3r) [inline src/guide/configure/intro.md:9]:
```suggestion
Read about the difference between underlying [types for configuration options](./configuration-types.md).
```

File: src/guide/configure/peer-configuration.md [hunk 3]
```diff
@@ -79,6 +92,14 @@ give you a (somewhat verbose) list of all available metrics gathered in
 Iroha. You might want to change this if you're having trouble gathering
 metrics using `prometheus`.
 
+## Genesis
+
+When you configure a peer, you have to provide private and public keys for the genesis account.
+
+You can do this via configuration file (`ACCOUNT_PUBLIC_KEY`, `ACCOUNT_PRIVATE_KEY`) or environment variables (`IROHA_GENESIS_ACCOUNT_PUBLIC_KEY`, `IROHA_GENESIS_ACCOUNT_PRIVATE_KEY`).
```
Comment (WRRicht3r) [inline src/guide/configure/peer-configuration.md:99]:
```suggestion
You can do this via the configuration file (`ACCOUNT_PUBLIC_KEY`, `ACCOUNT_PRIVATE_KEY`) or environment variables (`IROHA_GENESIS_ACCOUNT_PUBLIC_KEY`, `IROHA_GENESIS_ACCOUNT_PRIVATE_KEY`).
```

**Review by WRRicht3r — CHANGES_REQUESTED** [review]

File: src/guide/diagrams-src/domains-example.puml [hunk 4]
```diff
@@ -20,7 +20,7 @@ account_alice *-- asset_rose : registered by
 domain_wonderland *-- account_mad_hatter : registered in
 
 object domain_narnia {
-  id = {name = "narnia"}
+  id = "narnia"
```
[the remark this answers is not in the document]
Comment (outoftardis) [reply src/guide/diagrams-src/domains-example.puml:23]: changed the example

**Review by appetrosyan — APPROVED** [review]

**Pull request merged**